--- origsrc/wxPython-src-2.8.8.0/build/aclocal/bakefile.m4	2008-05-31 01:03:55.000000000 -0500
+++ src/wxPython-src-2.8.8.0/build/aclocal/bakefile.m4	2008-06-29 17:37:08.390625000 -0500
@@ -424,8 +424,8 @@
 
       *-*-cygwin* | *-*-mingw32* )
         PIC_FLAG=""
-        SHARED_LD_CC="\$(CC) -shared -o"
-        SHARED_LD_CXX="\$(CXX) -shared -o"
+        SHARED_LD_CC="\$(CC) -shared -Wl,--enable-auto-image-base -o"
+        SHARED_LD_CXX="\$(CXX) -shared -Wl,--enable-auto-image-base -o"
         WINDOWS_IMPLIB=1
       ;;
 
@@ -659,7 +659,7 @@
 
     LDFLAGS_GUI=
     case ${BAKEFILE_HOST} in
-        *-*-cygwin* | *-*-mingw32* )
+        *-*-mingw32* )
         LDFLAGS_GUI="-mwindows"
     esac
     AC_SUBST(LDFLAGS_GUI)
--- origsrc/wxPython-src-2.8.8.0/configure.in	2008-06-13 23:18:21.000000000 -0500
+++ src/wxPython-src-2.8.8.0/configure.in	2008-06-29 17:37:08.406250000 -0500
@@ -3360,7 +3360,7 @@
                     dnl we need to cache GTK_CFLAGS and GTK_LIBS for the
                     dnl subsequent runs
                     wx_cv_cflags_gtk=$GTK_CFLAGS
-                    wx_cv_libs_gtk=`echo $GTK_LIBS | sed -e 's/ -l[[^ ]]*cairo[[^ ]]*//g'`
+                    wx_cv_libs_gtk=$GTK_LIBS
                 fi
             ]
         )
