--- origsrc/wxPython-src-2.8.8.0/configure.in	2008-06-13 23:18:21.000000000 -0500
+++ src/wxPython-src-2.8.8.0/configure.in	2008-06-29 17:37:08.406250000 -0500
@@ -4072,13 +4072,21 @@
         dnl This is not ideal we really ough to use the unixodbc-config
         dnl  or iodbc-config if they exist.
 
-        AC_CHECK_HEADER([sql.h], [found_sql_h=1],, [ ])
+        AC_CHECK_HEADER([sql.h], [found_sql_h=1], , 
+        [ 
+            #ifdef __CYGWIN__
+            #include <windows.h>
+            #endif
+        ])
         if test "x$found_sql_h" = "x1" ; then
             AC_CHECK_LIB(iodbc, SQLAllocEnv, ODBC_LINK=" -liodbc",
             [
                 AC_CHECK_LIB(unixodbc, SQLAllocEnv, ODBC_LINK=" -lunixodbc",
                 [
-                    AC_CHECK_LIB(odbc, SQLAllocEnv, ODBC_LINK=" -lodbc")
+                    AC_CHECK_LIB(odbc, SQLAllocEnv, ODBC_LINK=" -lodbc",
+                    [
+                        AC_CHECK_LIB(odbc32, SQLAllocEnv, ODBC_LINK=" -lodbc32")
+                    ])
                 ])
             ])
         fi
--- origsrc/wxPython-src-2.8.8.0/include/wx/db.h	2007-07-04 14:24:57.000000000 -0500
+++ src/wxPython-src-2.8.8.0/include/wx/db.h	2008-06-29 17:37:08.421875000 -0500
@@ -59,6 +59,9 @@
 #else
     #if defined(__WINDOWS__) && ( defined(HAVE_W32API_H) || defined(__BORLANDC__)  || defined (__DMC__))
         #include "wx/msw/wrapwin.h"
+    #elif defined(__CYGWIN__)
+        #define WIN32_LEAN_AND_MEAN
+        #include <windows.h>
     #endif
     extern "C" {
     #if defined(wxUSE_BUILTIN_IODBC) && wxUSE_BUILTIN_IODBC
