--- origsrc/wxPython-src-2.8.8.0/include/wx/fontutil.h	2007-11-12 13:33:30.000000000 -0600
+++ src/wxPython-src-2.8.8.0/include/wx/fontutil.h	2008-06-29 17:37:08.421875000 -0500
@@ -224,12 +224,12 @@
 // translate a wxFontEncoding into native encoding parameter (defined above),
 // returning true if an (exact) macth could be found, false otherwise (without
 // attempting any substitutions)
-extern bool wxGetNativeFontEncoding(wxFontEncoding encoding,
+WXDLLIMPEXP_CORE bool wxGetNativeFontEncoding(wxFontEncoding encoding,
                                     wxNativeEncodingInfo *info);
 
 // test for the existence of the font described by this facename/encoding,
 // return true if such font(s) exist, false otherwise
-extern bool wxTestFontEncoding(const wxNativeEncodingInfo& info);
+WXDLLIMPEXP_CORE bool wxTestFontEncoding(const wxNativeEncodingInfo& info);
 
 // ----------------------------------------------------------------------------
 // font-related functions (X and GTK)
--- origsrc/wxPython-src-2.8.8.0/include/wx/gtk/gnome/gprint.h	2007-07-04 14:24:57.000000000 -0500
+++ src/wxPython-src-2.8.8.0/include/wx/gtk/gnome/gprint.h	2008-06-29 19:23:01.406250000 -0500
@@ -191,7 +191,7 @@
 // wxGnomePrintDC
 //-----------------------------------------------------------------------------
 
-class wxGnomePrintDC: public wxDC
+class WXDLLIMPEXP_CORE wxGnomePrintDC: public wxDC
 {
 public:
     wxGnomePrintDC( const wxPrintData& data );
--- origsrc/wxPython-src-2.8.8.0/include/wx/gtk/private.h	2007-07-10 10:31:49.000000000 -0500
+++ src/wxPython-src-2.8.8.0/include/wx/gtk/private.h	2008-06-29 17:37:08.437500000 -0500
@@ -33,7 +33,7 @@
     #include "wx/font.h"
 
     // convert the text in given encoding to UTF-8 used by wxGTK
-    WXDLLIMPEXP_CORE extern wxCharBuffer
+    WXDLLIMPEXP_CORE wxCharBuffer
     wxConvertToGTK(const wxString& s,
                    wxFontEncoding enc = wxFONTENCODING_SYSTEM);
 
@@ -111,7 +111,7 @@
 #include <gdk/gdktypes.h>
 
 // Returns stock accelerator modifier and key code for the given ID
-WXDLLEXPORT bool wxGetStockGtkAccelerator(const char *id, GdkModifierType *mod, guint *key);
+WXDLLIMPEXP_CORE bool wxGetStockGtkAccelerator(const char *id, GdkModifierType *mod, guint *key);
 #endif
 
 #endif // _WX_GTK_PRIVATE_H_
--- origsrc/wxPython-src-2.8.8.0/include/wx/unix/mimetype.h	2007-07-04 14:24:59.000000000 -0500
+++ src/wxPython-src-2.8.8.0/include/wx/unix/mimetype.h	2008-06-29 17:37:08.453125000 -0500
@@ -21,7 +21,7 @@
 WX_DEFINE_ARRAY_PTR(wxMimeTypeCommands *, wxMimeCommandsArray);
 
 // this is the real wxMimeTypesManager for Unix
-class WXDLLEXPORT wxMimeTypesManagerImpl
+class WXDLLIMPEXP_BASE wxMimeTypesManagerImpl
 {
 public:
     // ctor and dtor
@@ -145,7 +145,7 @@
 
 
 
-class WXDLLEXPORT wxFileTypeImpl
+class WXDLLIMPEXP_BASE wxFileTypeImpl
 {
 public:
     // initialization functions
--- origsrc/wxPython-src-2.8.9.1/include/wx/utils.h	2008-05-11 23:26:45.000000000 -0500
+++ src/wxPython-src-2.8.9.1/include/wx/utils.h	2009-02-10 14:17:48.218750000 -0600
@@ -692,7 +692,7 @@ void WXDLLEXPORT wxGetMousePosition( int
 // ----------------------------------------------------------------------------
 
 #ifdef __WXGTK__
-    void *wxGetDisplay();
+    WXDLLIMPEXP_CORE void *wxGetDisplay();
 #endif
 
 #ifdef __X__
--- origsrc/wxPython-src-2.8.9.2/include/wx/wxchar.h	2008-02-29 14:32:16.000000000 -0600
+++ src/wxPython-src-2.8.9.2/include/wx/wxchar.h	2009-03-30 18:40:15.843750000 -0500
@@ -1037,16 +1037,16 @@ WXDLLIMPEXP_BASE bool wxOKlibc(); /* for
         either because we don't have them at all or because they don't have the
         semantics we need
      */
-    int wxScanf( const wxChar *format, ... ) ATTRIBUTE_PRINTF_1;
-    int wxSscanf( const wxChar *str, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
-    int wxFscanf( FILE *stream, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
-    int wxVsscanf( const wxChar *str, const wxChar *format, va_list ap );
-    int wxPrintf( const wxChar *format, ... ) ATTRIBUTE_PRINTF_1;
-    int wxSprintf( wxChar *str, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
-    int wxFprintf( FILE *stream, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
-    int wxVfprintf( FILE *stream, const wxChar *format, va_list ap );
-    int wxVprintf( const wxChar *format, va_list ap );
-    int wxVsprintf( wxChar *str, const wxChar *format, va_list ap );
+    WXDLLIMPEXP_BASE int wxScanf( const wxChar *format, ... ) ATTRIBUTE_PRINTF_1;
+    WXDLLIMPEXP_BASE int wxSscanf( const wxChar *str, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
+    WXDLLIMPEXP_BASE int wxFscanf( FILE *stream, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
+    WXDLLIMPEXP_BASE int wxVsscanf( const wxChar *str, const wxChar *format, va_list ap );
+    WXDLLIMPEXP_BASE int wxPrintf( const wxChar *format, ... ) ATTRIBUTE_PRINTF_1;
+    WXDLLIMPEXP_BASE int wxSprintf( wxChar *str, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
+    WXDLLIMPEXP_BASE int wxFprintf( FILE *stream, const wxChar *format, ... ) ATTRIBUTE_PRINTF_2;
+    WXDLLIMPEXP_BASE int wxVfprintf( FILE *stream, const wxChar *format, va_list ap );
+    WXDLLIMPEXP_BASE int wxVprintf( const wxChar *format, va_list ap );
+    WXDLLIMPEXP_BASE int wxVsprintf( wxChar *str, const wxChar *format, va_list ap );
 #endif /* wxNEED_PRINTF_CONVERSION */
 
 /* these 2 can be simply mapped to the versions with underscore at the end */
--- origsrc/wxPython-src-2.8.9.2/src/aui/dockart.cpp	2008-09-06 19:09:09.000000000 -0500
+++ src/wxPython-src-2.8.9.2/src/aui/dockart.cpp	2009-03-31 02:23:38.828125000 -0500
@@ -71,7 +71,7 @@ unsigned char wxAuiBlendColour(unsigned 
 // or lightens a color, based on the specified percentage
 // ialpha of 0 would be completely black, 100 completely white
 // an ialpha of 100 returns the same colour
-wxColor wxAuiStepColour(const wxColor& c, int ialpha)
+WXEXPORT wxColor wxAuiStepColour(const wxColor& c, int ialpha)
 {
     if (ialpha == 100)
         return c;
--- origsrc/wxPython-src-2.8.8.0/src/gtk/utilsgtk.cpp	2007-11-12 13:33:31.000000000 -0600
+++ src/wxPython-src-2.8.8.0/src/gtk/utilsgtk.cpp	2008-06-29 17:37:08.500000000 -0500
@@ -59,6 +59,8 @@
     #include "X11/SM/SMlib.h"
 #endif
 
+#include "wx/gtk/private.h"
+
 //-----------------------------------------------------------------------------
 // data
 //-----------------------------------------------------------------------------
